src/compiler/checker.ts

```diff
@@ -29763,6 +29776,13 @@ namespace ts {
             }
             return false;
         }
+
+        function getEffectiveConstraintOfTypeParameter(node: TypeParameterDeclaration): TypeNode | undefined {

```

Comment:
Can we have this function exposed as part of the public API to align with getEffectiveTypeParameterDeclarations?